File: app/src/main/java/com/cbr/behancesampleapp/network/BehanceRepository.java
package com.cbr.behancesampleapp.network;

import com.cbr.behancesampleapp.BuildConfig;
import com.cbr.behancesampleapp.model.BehanceSinlgeUserReponse;
import com.cbr.behancesampleapp.model.BehanceUser;
import com.cbr.behancesampleapp.model.BehanceUserResponse;

import java.util.Map;

import io.reactivex.Observable;
import retrofit2.Response;

/**
 * Created by dimitrios on 25/08/2017.
 */
public class BehanceRepository extends DataRepository {

	private BehanceApiService mBehanceApiService;

	public BehanceRepository(BehanceApiService service) {
		this.mBehanceApiService = service;
	}

	public Observable<BehanceUserResponse> getUsers(Map<String, Object> params) {
		return mBehanceApiService.getUsers(params).compose(this.<BehanceUserResponse>applySchedulers());
	}

	public Observable<BehanceSinlgeUserReponse> getUserById(String userId) {
		return this.mBehanceApiService.getUserById(userId, BuildConfig.API_KEY).compose(this.<BehanceSinlgeUserReponse>applySchedulers());
	}

}

File: app/src/main/java/com/cbr/behancesampleapp/domain/network/repository/BehanceRepository.kt
package com.cbr.behancesampleapp.domain.network.repository

import com.cbr.behancesampleapp.BuildConfig
import com.cbr.behancesampleapp.domain.network.BehanceApiService
import com.cbr.behancesampleapp.model.BehanceListResponse
import com.cbr.behancesampleapp.model.BehanceProject
import com.cbr.behancesampleapp.model.BehanceUser
import com.cbr.behancesampleapp.model.BehanceUserResponse
import io.reactivex.Observable

/** Created by dimitrios on 25/08/2017. */
class BehanceRepository(private val mBehanceApiService: BehanceApiService) : DataRepository() {

    fun getUsers(params: MutableMap<String, Any>): Observable<BehanceListResponse<BehanceUser>> {
        return mBehanceApiService.getUsers(params).compose(this.applySchedulers())
    }

    fun getUserById(userId: String): Observable<BehanceUserResponse> {
        return this.mBehanceApiService.getUserById(userId, BuildConfig.API_KEY).compose(this.applySchedulers())
    }

    fun getProjects(params: MutableMap<String, Any>): Observable<BehanceListResponse<BehanceProject>> {
        return mBehanceApiService.getProjects(params).compose(this.applySchedulers())
    }

}